mittab/libs/data_export/tab_card.py

```diff
@@ -0,0 +1,350 @@
+                side,
+                get_victor_label(round_obj.victor, side),
+                opponent.display_backend,
+                " - ".join(j.name for j in round_obj.judges.all()),
+                (float(dstat1.speaks), float(dstat1.ranks)),
+                (float(dstat2.speaks), float(dstat2.ranks)),
+                (float(speaks_rolling), float(ranks_rolling)),
+            ]
+
+    for i in range(cur_round - 1):
+        # Don't fill in totals for incomplete rounds
+        if round_stats[i][6] == blank and len(round_stats)>=i+2 and blank not in round_stats[i + 1][:5]:
+            round_stats[i][6] = (0, 0) if i == 0 else round_stats[i - 1][6]
+
+    try:
+        bye_round = Bye.objects.get(bye_team=team).round_number
+    except Bye.DoesNotExist:
+        bye_round = None
+
+    return {
+        "team_school": team.school,
+        "debater_1": deb1.name,
+        "debater_1_status": Debater.NOVICE_CHOICES[deb1.novice_status][1],
+        "debater_2": deb2.name,
+        "debater_2_status": Debater.NOVICE_CHOICES[deb2.novice_status][1],
+        "round_stats": round_stats,
+        "d1st": tot_speaks_deb(deb1),
+        "d1rt": tot_ranks_deb(deb1),
+        "d2st": tot_speaks_deb(deb2),
+        "d2rt": tot_ranks_deb(deb2),
+        "ts": tot_speaks(team),
+        "tr": tot_ranks(team),
+        "bye_round": bye_round,
+    }
+
+
+def csv_tab_cards(writer):
+    # Write the CSV header row
+    header = [
+        "Team Name",
+        "School",
+        "Round id",
+        "Round",
+        "Gov/Opp",
+        "Win/Loss",
+        "Opponent",
+        "Chair",
+        "Wing(s)",
+        "Debater 1",
+        "N/V",
+        "Speaks",
+        "Ranks",
+        "Debater 2",
+        "N/V",
+        "Speaks",
+        "Ranks",
+        "Total Speaks",
+        "Total Ranks",
+    ]
+    writer.writerow(header)
+
+    teams = Team.objects.prefetch_related(
+        "school", "debaters", "gov_team", "opp_team")
+    total_rounds = TabSettings.objects.get(key="tot_rounds").value
+
+    for team in teams:
+        debaters = list(team.debaters.all())
+        deb1 = debaters[0]
+        deb1_status = Debater.NOVICE_CHOICES[deb1.novice_status][1][0]
+        iron_man = len(debaters) < 2
+        if iron_man:
+            deb2 = deb1
+            deb2_status = deb1_status
+        else:
+            deb2 = debaters[1]
+            deb2_status = Debater.NOVICE_CHOICES[deb2.novice_status][1][0]
+
+        rounds = list(team.gov_team.all()) + list(team.opp_team.all())
+        round_data = [{}] * total_rounds
+
+        for round_obj in rounds:
+            side = GOV if round_obj.gov_team == team else OPP
+            result = get_victor_label(round_obj.victor, side)
+            opponent = (
+                round_obj.opp_team if round_obj.gov_team == team else round_obj.gov_team
+            )
+            opponent_name = opponent.get_or_create_team_code() if opponent else "BYE"
+            chair = round_obj.chair.name
```

Comment:
e.g. this should be `round_obj.chair if round_obj.chair.name is not None else None`